frontend/src/components/admin/AssignedStoryTranslationsTable.tsx

```diff
@@ -0,0 +1,176 @@
+import React, { useState } from "react";
+import { Icon } from "@chakra-ui/icon";
+import { MdDelete } from "react-icons/md";
+import {
+  Badge,
+  Button,
+  Table,
+  Thead,
+  Tbody,
+  Tr,
+  Th,
+  Td,
+  IconButton,
+} from "@chakra-ui/react";
+import { StoryTranslation } from "../../APIClients/queries/StoryQueries";
+import { getLevelVariant } from "../../utils/StatusUtils";
+import convertStageTitleCase from "../../utils/StageUtils";
+import { convertLanguageTitleCase } from "../../utils/LanguageUtils";
+
+export type AssignedStoryTranslationsTableProps = {
+  storyTranslations: StoryTranslation[];
+  setStoryTranslations: (newStoryTranslations: StoryTranslation[]) => void;
+  userId: number;
+};
+
+const AssignedStoryTranslationsTable = ({
+  storyTranslations,
+  setStoryTranslations,
+  userId,
+}: AssignedStoryTranslationsTableProps) => {
+  const [isAscendingTitle, setIsAscendingTitle] = useState(true);
+  const [isAscendingRole, setIsAscendingRole] = useState(true);
+  const [isAscendingLanguage, setIsAscendingLanguage] = useState(true);
+  const [isAscendingStage, setIsAscendingStage] = useState(true);
+
+  const getRole = (translation: StoryTranslation) => {
+    if (userId === translation.translatorId) {
+      return "Translator";
+    }
+    if (userId === translation.reviewerId) {
+      return "Reviewer";
+    }
+    return "Error";
+  };
+
+  const titleSort = (t1: StoryTranslation, t2: StoryTranslation) =>
+    isAscendingTitle
+      ? t1.title.localeCompare(t2.title)
+      : t2.title.localeCompare(t1.title);
+
+  const roleSort = (t1: StoryTranslation, t2: StoryTranslation) =>
+    isAscendingRole
+      ? getRole(t1).localeCompare(getRole(t2))
+      : getRole(t2).localeCompare(getRole(t1));
+
+  const languageSort = (t1: StoryTranslation, t2: StoryTranslation) =>
+    isAscendingLanguage
+      ? t1.language.localeCompare(t2.language)
+      : t2.language.localeCompare(t1.language);
+
+  const stageSort = (t1: StoryTranslation, t2: StoryTranslation) =>
+    isAscendingStage
+      ? t1.stage.localeCompare(t2.stage)
+      : t2.stage.localeCompare(t1.stage);
+
+  const sort = (field: String) => {
+    const newStoryTranslations = [...storyTranslations];
+    switch (field) {
+      case "title": {
+        setIsAscendingTitle(!isAscendingTitle);
+
```

Comment:
nit: this empty line bothers me for some reason LOL did this come from the linter?